### ISSUE TITLE
Change default permission for sharing by link or mail

### ISSUE THREAD

**Issue by Hativ** [nextcloud/server#25713]
**Is your feature request related to a problem? Please describe.**

We always allow uploading and editing when we share a file or folder by link or mail. Therefore it's annoying to change the permissions for every shared link to it, because default setting is "read only".

**Describe the solution you'd like**
I would like to be able to set the default permission for sharing by link and mail to "Allow uploading and editing" for the whole Nextcloud installation.

**Additional context**
![image](https://user-images.githubusercontent.com/12894547/108404625-5f666c80-7220-11eb-966e-1bb700dfcb1d.png)

**Comment by szaimen** [comment]
This sjould be already possible in the admin sharing settings 
![Screenshot_20210218-214325_Microsoft Remote Desktop](https://user-images.githubusercontent.com/42591237/108419201-8463db00-7232-11eb-8b4a-9f5de48ebdfa.png)

**Comment by Hativ** [comment]
@szaimen The settings have no effect to share by link or mail.

I configured it like this:
![image](https://user-images.githubusercontent.com/12894547/108466998-d46b8d80-7284-11eb-89ce-be0dd432bef4.png)

The standard permissions when I share by link or mail are still "read only". Nextcloud 20.0.7

According to documentation the settings are only for user-shares, not share by link or mail.

> With Default share permissions you are able to set the default permissions for user-shares (Create, Change, Delete and Reshare) without forcing them.
https://docs.nextcloud.com/server/stable/admin_manual/configuration_files/file_sharing_configuration.html

**Comment by szaimen** [comment]
> I would like to be able to set the default permission for sharing by link and mail to "Allow uploading and editing" for the whole Nextcloud installation.

cc @nextcloud/designers does this make sense?

**Comment by nimisha-vijay** [comment]
Yes, we can have another setting under "Allow users to share via link and mail" which says "Default share permissions", with the 3 options under it. 
Also might want to change the currently existing "Default share permissions" to "Default permissions for user shares" so that there's no confusion.

**Comment by Crocmagnon** [comment]
I'm also interested in this feature 👍🏻

**Comment by borisbaldassari** [comment]
+1
